src/frontend/sharedComponents/BGMapCard.tsx

```diff
@@ -23,86 +29,258 @@ const m = defineMessages({
     defaultMessage: "Unnamed Style",
     description: "The name for the default map style",
   },
+  waitingForImport: {
+    id: "sharedComponents.BGMapCard.waitingForImport",
+    defaultMessage: "Waiting for import…",
+    description: "Message indicating that import is waiting to start",
+  },
+  importInProgress: {
+    id: "sharedComponents.BGMapCard.importInProgress",
+    defaultMessage: "Import in progress…",
+    description: "Message about the import being in progress",
+  },
 });
 
-// ToDo: API calls to get styleURL, zoom level, center coordinate, etc.
+type ImportStatus =
+  | { status: "error"; soFar?: number; total?: number }
+  | {
+      status: "progress" | "complete";
+      soFar: number;
+      total: number;
+    }
+  | null;
+
+function shouldShowProgressBar(
+  importStatus: ImportStatus
+): importStatus is {
+  status: "progress" | "complete";
+  soFar: number;
+  total: number;
+} | null {
+  return (
+    importStatus === null ||
```

Comment:
yeah it's kind of confusing and maybe there's a better way to represent the state explicit status strings, but basically when the `importStatus` is null, it could mean that we've started an import but haven't received any events back yet. In that case, we want to show the progress bar because it will be in the indeterminate state.

the reason I'm using a null value at all is because if the map's already been imported some time before e.g. yesterday, then there's no need to have an importStatus associated for it, hence null.

would it be helpful to add an import status that represents this "waiting" situation? e.g. extends the `ImportStatus` type so that there's a variant like `{ status: 'waiting' }`?

Reply:
Yeah, it can't hurt to have that explicit status of `waiting`. And looking through the code, it would simplify some of your null check logic

Reply:
see 4012d1da791f78ebcea8a7a0f74189ac938e6f37

Reply:
did additional work on this via 8bcad51dd905de3f2a7c9e21ba007b9aeb7fe0a7 and think it's much easier to follow now. basically removes the `null` variant and updates the wording of `"waiting"` to `"idle"`